metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -74,9 +71,20 @@ case class BuildServerConnection(
   def mainClasses(
       params: ScalaMainClassesParams
   ): CompletableFuture[ScalaMainClassesResult] = {
-    // TODO use server.buildTargetScalaMainClasses when bloop releases version supporting mainClasses
-    CompletableFuture.completedFuture(
-      new ScalaMainClassesResult(Collections.emptyList())
+    register(server.buildTargetScalaMainClasses(params))
+  }
+
+  def testClasses(
+      params: ScalaTestClassesParams
+  ): CompletableFuture[ScalaTestClassesResult] = {
+    register(server.buildTargetScalaTestClasses(params))
+  }
+
+  def startDebugSession(params: DebugSessionParams): CompletableFuture[URI] = {
```

Comment:
Would it make sense to add a guard here to make sure we only send this request to an up-to-date Bloop server that supports debugging? I wonder if we should introduce a debug capability to BSP during the `build/initialize` handshake 🤔

Reply:
What do you mean by "up to date Bloop server"?
I think it would be much simpler to just let the bsp server tell us by responding"I don't support debugging because: A or B or C" (which already happens, for example when bloop runs on jre and not on jdk). The cost is relatively low, because running is a seldom triggered action. 

Of course, in principle you are right, the debug capability should also be included in the capabilities of the server but in this case the gain would be minuscule

Reply:
I see, thanks for the explanation. For now this is fine. It would be nice to introduce a "debug capability" in BSP so that we only send debug requests to build servers that support it. Currently, Metals even ignores the "compile capability".